src/VisualStudio/Core/Def/LanguageClient/VisualStudioLspWorkspaceRegistrationService.cs

```diff
@@ -4,22 +4,19 @@
 
 using System;
 using System.Composition;
-using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Host.Mef;
 
 namespace Microsoft.CodeAnalysis.LanguageServer;
 
 [Export(typeof(LspWorkspaceRegistrationService)), Shared]
-internal class VisualStudioLspWorkspaceRegistrationService : LspWorkspaceRegistrationService
+internal sealed class VisualStudioLspWorkspaceRegistrationService : LspWorkspaceRegistrationService
 {
     [ImportingConstructor]
     [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
     public VisualStudioLspWorkspaceRegistrationService()
     {
     }
 
-    public override string GetHostWorkspaceKind() => WorkspaceKind.Host;
-
     public override void Register(Workspace? workspace)
```

Comment:
I think now that we've removed the `LspMiscellaneousFilesWorkspace` from VS (I'm assuming its not created at all anymore there) - I think we can just export a single LspWorkspaceRegistration service and remove the VS and VSMac ones